packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -289,47 +330,81 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 							this.hostPolicy.concurrentSnapshotFetch &&
 							!this.hostPolicy.summarizerClient
 						) {
-							const networkSnapshotP = this.fetchSnapshot(
-								hostSnapshotOptions,
-								scenarioName,
-							);
-
 							// Ensure that failures on both paths are ignored initially.
-							// I.e. if cache fails for some reason, we will proceed with network result.
-							// And vice versa - if (for example) client is offline and network request fails first, we
+							// I.e. if cache fails for some reason, we will proceed with prefetched/network result.
+							// And vice versa - if (for example) client is offline and prefetched/network request fails first, we
 							// do want to attempt to succeed with cached data!
-							const promiseRaceWinner = await promiseRaceWithWinner([
-								cachedSnapshotP.catch(() => undefined),
-								networkSnapshotP.catch(() => undefined),
-							]);
-							retrievedSnapshot = promiseRaceWinner.value;
-							method = promiseRaceWinner.index === 0 ? "cache" : "network";
-
-							if (retrievedSnapshot === undefined) {
-								// if network failed -> wait for cache ( then return network failure)
-								// If cache returned empty or failed -> wait for network (success of failure)
-								if (promiseRaceWinner.index === 1) {
-									retrievedSnapshot = await cachedSnapshotP;
-									method = "cache";
+							if (prefetchedSnapshotP) {
```

Comment:
I got lost here on why we need to much change :) It feels like we should simply treat prefetchedSnapshotP and networkSnapshotP the same way. I.e. I do not think there is that much value to record differences between "network" and "prefetched" - after all the difference is only that network call started a bit sooner. If you really want to capture that, I'd rather record (in non-persistent cache) when it started, and record delta as a property on ObtainSnapshot event.
The rest of the logic seems like should not change at all.

Reply:
Changed the logic.